### PR TITLE
Tests: Add PHP 8.4 and move PCOV to PHP 8.2

### PR DESCRIPTION
**Pull request by acicovic**
## Description
This PR adds PHP 8.4 to our unit and integration test matrices in experimental mode, as PHP 8.4 is on its [merry way to release](https://wiki.php.net/todo/php84) towards the end of November.

It also moves PCOV from using PHP 8.1 to 8.2, which is currently the most popular PHP version in the 8.x branch according to [WordPress Stats](https://wordpress.org/about/stats/).

## Motivation and context
- Keep our testing current
- Catch issues early if they occur.

## How has this been tested?
- No functionality changes.
- GitHub testing workflows should start using PHP 8.4 and PCOV with PHP 8.2.

## Summary by CodeRabbit

- **New Features**
	- Enhanced integration and unit testing workflows to support additional PHP versions (8.1, 8.3, and 8.4).
	- Improved flexibility in handling request parameters across various methods.

- **Bug Fixes**
	- Adjusted conditions for running integration tests to exclude PHP 8.2.

- **Refactor**
	- Updated method signatures to allow nullable parameters for improved type safety and flexibility.
	- Added a constant to indicate when integration tests are running.

### CODE REVIEW

**Comment by acicovic** [comment]
This work is based on the `refactor/external-services` branch.

The PHP 8.4 deprecation errors that GitHub was flagging have been taken care of. We do have test errors which is expected, since tests aren't updated yet.

**Comment by acicovic** [comment]
After the integration test fixes on the base branch, we do have integration test failures that are specific to PHP 8.4.

**Comment by acicovic** [comment]
Opening this PR though we're getting PHP 8.4 failures on integration tests. To be discussed what we do with that.

**Review by vaurdan — APPROVED** [review]

File: .github/workflows/integration-tests.yml [hunk 1]
```diff
@@ -92,17 +83,17 @@ jobs:
...
 
-      - name: Run integration tests (multi site)
-        if: ${{ matrix.php != 8.0 && ! matrix.experimental }}
+      - name: Run integration tests (multisite)
+        if: ${{ matrix.php != 8.2 && ! matrix.experimental }}
         run: composer testwp-ms --no-interaction
 
       - name: Run integration tests (multisite site with code coverage)
-        if: ${{ matrix.php == 8.1 }}
+        if: ${{ matrix.php == 8.2 }} # To prevent unneeded test runs, use this version in the above if clauses.
```
[the remark this answers is not in the document]
Comment (acicovic) [reply .github/workflows/integration-tests.yml]: @coderabbitai, can you check that the changes are correct?

**Pull request merged**